designdocs/extended_token_lock.md

```diff
@@ -1,121 +1,153 @@
+of a new `boosting program` types to handle creation of programs with new structures. The objective is to support tokens locked with specific
 thawing behavior, with such tokens participating in Provider Boosting at a distinct reward rate.
 
-## 📂 `pallet-delayed_thaw` - Token Locking Mechanism
+### Key Terms
 
-This new pallet manages the locking and gradual thawing of tokens. It does not handle reward logic but contributes to
-reward calculation via a custom implementation of the `RewardsProvider` trait.
+- **Extended Boosting Program**: New reward program with distinct rules.
+- **Precipitating Tokenomic Event (PTE)**: Governance-triggered event that starts unlock schedule.
+- **Pre-PTE Freeze**: Lock phase before PTE.
+- **Post-PTE Freeze**: Lock phase after PTE but before thawing starts.
+- **Extended Thaw**: Gradual unlock phase after freeze ends.
+- **Expired Program**: All tokens can be unlocked; normal rewards resume.
+
+### Extended Boosting Phases
+
+| Phase            | Can Join | Can Unstake | Unstake Amount | Stake Reward     |
+|------------------|----------|-------------|----------------|------------------|
+| Pre-PTE          | ✅       | 🚫          | 0%             | Extended Rewards |
+| Post-PTE         | ✅       | 🚫          | 0%             | Extended Rewards |
+| Extended Thaw    | 🚫       | ✅          | Formula-Based  | Extended Rewards |
+| Expired Program  | 🚫       | ✅          | 100%           | Default Rewards  |
+| Failsafe Trigger | 🚫       | ✅          | 100%           | Default Rewards  |
+
+## 📂 Capacity Pallet Changes
 
 ### Storage and Constants:
 
 ```rust
+pub enum StakingType {
+    MaximumCapacity,
+    ProviderBoost,
+    ProviderExtendedBoost, // NEW
+}
+
+pub const EXTENDED_BOOST_FAILSAFE_UNLOCK_BLOCK_NUMBER: BlockNumber = <some constant>;
+
 #[pallet::storage]
 pub type PrecipitatingEventBlockNumber<T: Config> = StorageValue<_, T::BlockNumber, OptionQuery>;
-
-pub const FAILSAFE_UNLOCK_BLOCK_NUMBER: BlockNumber = < some constant>;
 ```
 
 - `PrecipitatingEventBlockNumber`: Set by governance to signal a precipitating event.
-- `FAILSAFE_UNLOCK_BLOCK_NUMBER`: Ensures unlock safety if governance fails to act.
+- `EXTENDED_BOOST_FAILSAFE_UNLOCK_BLOCK_NUMBER`: Block after which full unlock is allowed if PTE does not occur.
+
+
+### Reward Parameters for Extended Boosting
+
+- `RewardPercentCap` from `Permill::from_parts(5_750);` to `Permill::from_parts(TBD);`
+
+#### Notes
+
+- This could be a new `RewardsProvider` implementation; however, it MUST NOT be possible to exceed the `RewardPoolPerEra` which a simple adjustment of the cap could enable.
 
 ### Thaw Parameters:
 
 ```rust
-pub const THAW_ERA_LENGTH: BlockNumber = < some constant>;
-pub const INITIAL_FREEZE_THAW_ERAS: u32 = < some constant>;
-pub const UNLOCK_THAW_ERAS: u32 = < some constant>;
+/// Number of epochs after the `PrecipitatingEventBlockNumber` that no unstaking is allowed
+pub const INITIAL_FREEZE_THAW_EPOCHS: u32 = < some constant>;
```

Comment:
this is based on Capacity Epochs (currently ~1 day long), not RewardEras?

Reply:
Sorry; terminology mixup. Reward eras is what I was going for--will update